src/AutoRest.CSharp/Input/SchemaUsageProvider.cs

```diff
@@ -18,7 +18,9 @@ public SchemaUsageProvider(CodeModel codeModel)
 
                 if (usage != null)
                 {
-                    Apply(objectSchema, (SchemaTypeUsage) Enum.Parse(typeof(SchemaTypeUsage), usage, true));
+                    var schemaTypeUsage = (SchemaTypeUsage)Enum.Parse(typeof(SchemaTypeUsage), usage, true);
+                    var recurse = schemaTypeUsage.HasFlag(SchemaTypeUsage.Converter) ? false : true;
```

Comment:
This is not correct, all other usages need to be applied recursively, only the SchemaTypeUsage.Converter shouldn't

Reply:
I think you need to change the `Apply` method and clear the `SchemaTypeUsage.Converter` flag after it was applied to the initial schema.

Reply:
Current changes also work fine without changing `Apply` method. Let me know if I'm missing something.